Fix Auto Advance waiting for audio when autoplay is disabled

jradchaima opened the pull request:
<!--- Please fill the necessary details below -->
## Purpose / Description
Auto Advance does not work when both "Don't play audio automatically" and "Wait for audio" are enabled 
* #17937

## Fixes
* Fixes #17937

## Approach
TheFile Abstractflashcardviewer.k contain automaticAnswerShouldWaitForAudio()  function 
https://github.com/ankidroid/Anki-Android/blob/main/AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt#L1309-L1312
this function  only checks "Wait for audio" but ignores autoplay, which means Auto Advance might still wait even if autoplay is disabled.Since this function doesn’t check autoplay, we need to fix it so Auto Advance behaves correctly.
```kotlin
suspend fun automaticAnswerShouldWaitForAudio(): Boolean = withCol {
    val card = currentCard ?: return@withCol false  // Prevents null crash
    val waitForAudio = decks.configDictForDeckId(card.did).waitForAudio
    val soundConfig = CardSoundConfig.create(this, card)  // Fetch autoplay setting

    return@withCol waitForAudio && soundConfig.autoplay  //  Only wait if autoplay is enabled
}
```
Since ReviewerViewModel.kt does not inherit from AbstractFlashcardViewer.kt, we need a reference to an instance of AbstractFlashcardViewer.then call the function through flashcardViewer
```kotlin
suspend fun shouldWaitForAudio(): Boolean {
    return flashcardViewer?.automaticAnswerShouldWaitForAudio() ?: false
}
```
then Modify AutoAdvance.kt so it correctly calls shouldWaitForAudio() from ReviewerViewModel.kt
```kotlin
suspend fun shouldWaitForAudio(): Boolean {
    return viewModel.shouldWaitForAudio()  
}
```

## How Has This Been Tested?

Instructions to reprodruce
1-(Optional) Confirm that "Auto Advance" feature is working in the selected deck in Reviewer
2-Long-press the deck in the main screen > "Deck options" >
"Audio" > set "Don't play audio automatically" enabled
(It is set to disabled by default)
3-"Auto Advance" > set "Wait for audio" enabled
(It is already set to enabled by default)
4-Tap "Save"
5-Tap the deck to open Reviewer
verfication:AutoAdvance behaves correctly

https://github.com/user-attachments/assets/7a281a54-b463-4d5c-8875-32dd48970299

tested on android device samsung

## Learning (optional, can help others)
_Describe the research stage_

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

BrayanDSO commented on AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/reviewer/ReviewerViewModel.kt:
```diff
@@ -509,4 +511,8 @@ class ReviewerViewModel(
                 title
             }
     }
+    suspend fun shouldWaitForAudio(): Boolean {
+        return flashcardViewer?.automaticAnswerShouldWaitForAudio() ?: false
+    }
```
this is completely separated from AbstractFlashcardViewer. Do you really know what you are trying to do?

BrayanDSO commented on AnkiDroid/src/main/java/com/ichi2/anki/ui/windows/reviewer/autoadvance/AutoAdvance.kt:
```diff
@@ -61,7 +61,9 @@ class AutoAdvance(
     private suspend fun answerAction() = settings.await().answerAction
 
-    suspend fun shouldWaitForAudio() = settings.await().waitForAudio
+    suspend fun shouldWaitForAudio(): Boolean {
+        return viewModel.shouldWaitForAudio()
+    }
```
ditto

jradchaima commented:
Hi,
I removed the unnecessary code that I added to AutoAdvance.kt and ReviewerViewModel.kt, and tested auto-advance when both 'Don't play audio automatically' and 'Wait for audio' are enabled. It works fine

BrayanDSO commented on AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:
```diff
@@ -1308,7 +1309,10 @@ abstract class AbstractFlashcardViewer :
     private suspend fun automaticAnswerShouldWaitForAudio(): Boolean =
         withCol {
-            decks.configDictForDeckId(currentCard!!.did).waitForAudio
+            val card = currentCard ?: return@withCol false // Prevents null crash
+            val waitForAudio = decks.configDictForDeckId(card.did).waitForAudio
+            val soundConfig = CardSoundConfig.create(this, card) // Fetch autoplay setting
+            return@withCol waitForAudio && soundConfig.autoplay //  Only wait if autoplay is enabled
```
the comments aren't necessary in this case
```suggestion
            val card = currentCard ?: return@withCol false
            val waitForAudio = decks.configDictForDeckId(card.did).waitForAudio
            val soundConfig = CardSoundConfig.create(this, card)
            return@withCol waitForAudio && soundConfig.autoplay
```

BrayanDSO commented on AnkiDroid/src/main/java/com/ichi2/anki/AbstractFlashcardViewer.kt:
```diff
@@ -1308,7 +1309,10 @@ abstract class AbstractFlashcardViewer :
 
     private suspend fun automaticAnswerShouldWaitForAudio(): Boolean =
         withCol {
-            decks.configDictForDeckId(currentCard!!.did).waitForAudio
+            val card = currentCard ?: return@withCol false
+            val waitForAudio = decks.configDictForDeckId(card.did).waitForAudio
+            val soundConfig = CardSoundConfig.create(this, card)
+            return@withCol waitForAudio && soundConfig.autoplay
         }
```
the method does more than what its name say it does. The issue probably isn't here, but in the calls for auto advance to run
